src/compiler/checker.ts

```diff
@@ -6835,6 +7028,23 @@ module ts {
                     });
                 }
             }
+            else if (node.kind === SyntaxKind.Decorator) {
+                let decorator = <Decorator>node;
+                switch (decorator.parent.kind) {
+                    case SyntaxKind.ClassDeclaration:
+                        args = [decorator.expression];
+                        break;
+                    case SyntaxKind.PropertyDeclaration:
+                        args = [decorator.expression, decorator.expression];
+                        break;
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.GetAccessor:
+                    case SyntaxKind.SetAccessor:
+                    case SyntaxKind.Parameter:
+                        args = [decorator.expression, decorator.expression, decorator.expression];

```

Comment:
I thought in the case of method or accessor, you want only two arguments if the signature has two parameters. Yet here you are passing three regardless. How does this not break the two parameter case?